python/hail/expr/expressions/utils.py

```diff
@@ -58,22 +59,24 @@ def analyze(caller, expr, expected_indices, aggregation_axes=set()):
             expected_agg_axes = expected_axes.union(aggregation_axes)
 
             for agg in aggregations:
+                assert isinstance(agg, Aggregation)
+                refs = get_refs(*agg.exprs)
                 agg_indices = agg.indices
                 agg_axes = agg_indices.axes
                 if agg_indices.source is not None and agg_indices.source is not expected_source:
                     errors.append(
                         ExpressionException(
                             'Expected an expression from source {}, found expression derived from {}'
                             '\n    Invalid fields: [{}]'.format(
-                                expected_source, source, ', '.join("'{}'".format(name) for name, _ in agg.refs)
+                                expected_source, source, ', '.join("'{}'".format(name) for name, _ in refs)
                             )))
 
                 # check for stray indices
                 unexpected_agg_axes = agg_axes - expected_agg_axes
                 if unexpected_agg_axes:
                     # one or more out-of-scope fields
                     bad_refs = []
-                    for name, inds in agg.refs:
+                    for name, inds in refs:
                         bad_axes = inds.axes.intersection(unexpected_agg_axes)
                         if bad_axes:
                             bad_refs.append((name, inds))
```

Comment:
Do you need an assert statement here same as in line 41.

Reply:
added